lib/active_merchant/billing/gateways/s5.rb

```diff
@@ -0,0 +1,236 @@
+        'authonly'  => 'CC.PA',
+        'capture'   => 'CC.CP',
+        'refund'    => 'CC.RF',
+        'void'      => 'CC.RV',
+      }
+
+      def initialize(options={})
+        requires!(options, :sender, :channel, :login, :password)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+        add_memo(post, options) if options[:memo]
+
+        commit('sale', post)
+      end
+
+      def refund(money, authorization, options={})
+        post = {}
+        post[:authorization] = authorization
+        add_invoice(post, money, options)
+        add_memo(post, options) if options[:memo]
+
+        commit('refund', post)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+        add_memo(post, options) if options[:memo]
+
+        commit('authonly', post)
+      end
+
+      def capture(money, authorization, options={})
+        post = {}
+        post[:authorization] = authorization
+        add_invoice(post, money, options)
+        add_memo(post, options) if options[:memo]
+
+        commit('capture', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+        post[:authorization] = authorization
+        add_memo(post, options) if options[:memo]
+
+        commit('void', post)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r(<Security.+), '\1[FILTERED]\2').
+          gsub(%r(<User login.+), '\1[FILTERED]\2').
+          gsub(%r((<Number>).+(</Number>)), '\1[FILTERED]\2').
+          gsub(%r((<Verification>).+(</Verification>)), '\1[FILTERED]\2')
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email]
+        post[:ip] = options[:ip]
+      end
+
+      def add_address(post, creditcard, options)
+        if address = options[:billing_address] || options[:address]
+          post[:address] = "#{address[:address1]} #{address[:address2]}"
+          post[:company] = address[:company].to_s
+          post[:phone]   = address[:phone].to_s.gsub(/[^0-9]/, '') || "0000000"
+          post[:zip]     = address[:zip].to_s
+          post[:city]    = address[:city].to_s
+          post[:country] = address[:country].to_s
+          post[:state]   = address[:state].blank?  ? 'n/a' : address[:state]
```

Comment:
I'd recommend using 

https://github.com/Shopify/active_merchant/blob/master/lib/active_merchant/empty.rb

since we're trying to eliminate the dependency on ActiveSupport.  Like this:

https://github.com/Shopify/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L6
https://github.com/Shopify/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L307